service/scaladsl/kafka/server/src/main/scala/com/lightbend/lagom/internal/scaladsl/broker/kafka/ScaladslRegisterTopicProducers.scala

```diff
@@ -82,11 +82,8 @@ class ScaladslRegisterTopicProducers(
                   projectionRegistryImpl
                 )
 
-              case other =>
-                log.warn {
-                  s"Unknown topic producer ${other.getClass.getName}. " +
-                    s"This will likely result in no events published to topic ${topicId.name} by service ${info.serviceName}."
-                }
+              case null => log.error(s"Expected an instance of ${classOf[ScalaMethodTopic[_]]}, but 'null' was passed",
+                  new NullPointerException(s"expected an instance of ${classOf[ScalaMethodTopic[_]]}, but 'null' was passed"))
```

Comment:
After checking it further, I think this is not yet what we need. 

The previous code was removed. We do need to have a `case other => `. This is needed to detect if an unsupported instance was passed. 

I suggest the following instead:

```scala 
case null =>
  val message = s"Expected an instance of ${classOf[TaggedOffsetTopicProducer[_, _]]}, but 'null' was passed"
  log.error(message, new NullPointerException(message))

case other =>
  log.warn {
    s"Unknown topic producer ${other.getClass.getName}. " +
      s"This will likely result in no events published to topic ${topicId.name} by service ${info.serviceName}."
  }
```

Note that I changed `ScalaMethodTopic` to `TaggedOffsetTopicProducer`. This is the instanced we are expecting.